### PR TITLE
Fix #182: Add missing performative logging for open/close and SASL

### LINKED ISSUE #182: No debug logging for open/close/SASL performatives
When trying to diagnose an AMQP connection problem turning up the logging to reveal the protocol trace is often a useful strategy.   However, I notice that logging of a few performatives such as Open/Close and SASL are missing.

Would it be possible to log these too?

### PR DESCRIPTION
**Pull request by k-wall**

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -609,9 +609,11 @@ func (c *conn) connWriter() {
 		// connection complete
 		case <-c.done:
 			// send close
+			close := &performClose{}
```
Comment (vcabbage) [inline conn.go]: Can you rename this so it doesn't shadow the `close` built in function?

Comment (k-wall) [reply conn.go]: Thanks for the review, change made.

**Comment by vcabbage** [comment]
Thanks for making that change. Having thought about this a bit more, perhaps these logs should only be printed at a high debug level, like 5. My reasoning is that the debug logs are commonly requested in GitHub issues and the SASL frames include authentication credentials. These debug lines print them out in hex, so it would be easy for someone unfamiliar with the protocol to overlook them when redacting private info.

Thoughts?

**Comment by k-wall** [comment]
Apache Qpid Proton has a really useful feature that allows a user to get a *complete* protocol trace without changing/rebuilding code.  This is achieved by setting the environment variable `PN_TRACE_FRM`.  This turns out to be really handy and is often the first thing is is requested when problem solving in AMQP powered networks.  I was going to suggest something analogous for this library (under a separate issue).  

So, with this in mind, rather than moving the logging of Sasl performatives to a different logging level, I'd suggest better just to obfuscate (say with asterisk characters) the contents of fields initial-response/response fields in their respective String() implementations?  Would this be acceptable?

Separately it would be nice if the library support scram-sha-256, which avoids the need for the plain text on the wire.

**Comment by vcabbage** [comment]
Obfuscating the credentials sounds good :+1:

**Comment by vcabbage** [comment]
Thanks!

**Pull request merged**